automation-ui/automation-base/src/main/java/com/yas/automation/base/page/BasePage.java

```diff
@@ -39,6 +39,35 @@ public void wait(Duration duration) {
         }
     }
 
+    public void waitWithRetry(Duration duration, Supplier<Boolean> booleanSupplier) {
```

Comment:
Hi @chanhnguyen6975 , i think you should use the waiting mechanism that provided by Selenium instead of using Thread.sleep. For example: within 5 seconds, if element "table tbody tr" is visible, Selenium will stop waiting immediately

```
final WebDriver chromeDriver = webDriverFactory.getChromeDriver();
WebDriverWait w = new WebDriverWait(chromeDriver, Duration.ofSeconds(5));
w.until(ExpectedConditions.presenceOfElementLocated (By.cssSelector("table tbody tr"))); 
```

Reply:
Hi anh @khanhtrand, 
Some elements on the page are already loaded but are not interactable. In that case, we can retrieve the WebElement object, but we cannot call methods like `WebElement.click(...)`, `WebElement.sendKeys(...)`, etc.

Therefore, the purpose of this method is to make the waiting conditional (pre-check if above method can invoke), not just to check if the element is present or not.

Let me know you ideas, suggestions. 😉

Reply:
I think `ExpectedConditions` supports many conditions including `elementToBeClickable(By locator)`

We can create a `SeleniumHelper` with a common method like

```
public <T> T waitForCondition(Duration duration, ExpectedCondition<T> condition) {
        WebDriverWait wait = new WebDriverWait(driver, duration);
        return wait.until(condition);
    }
```
Call it like below
```
WebElement button = helper.waitForCondition(
    Duration.ofSeconds(5), 
    ExpectedConditions.elementToBeClickable(By.id("button_id"))
);
button.click();
```

Reply:
Yah, thanks anh Khanh, 
Let us check if we can use it for our case.